src/function/handler.ts

```diff
@@ -1,12 +1,15 @@
 import type { Context } from './context'
 import type { Event } from './event'
-import type { Response } from './response'
+import type { Response, BuilderResponse } from './response'
 
-export interface HandlerCallback {
+export interface HandlerCallback<TResponse> {
```

Comment:
Can we find a more descriptive name for `TResponse`? Maybe `ResponseType`?

Also, should it default to `Response`? Otherwise, I think this will be a breaking change for anyone using the `HandlerCallback` type?

Reply:
`ResponseType` sounds good! And good call on HandlerCallback defaulting to response as well, I'll add that!